generator/src/lib.rs

```diff
@@ -1755,7 +1773,7 @@ pub fn derive_setters(
                     //                 next chain
                     let last_next = ptr_chain_iter(next).last().unwrap();
                     (*last_next).p_next = self.inner.p_next as _;
```

Comment:
Not sure about this `as _` - it seems we have some `inner.p_next` that are `*mut c_void`, others are `*const c_void`. The target type of `BaseOutStructure::p_next` (LHS here) is always `*mut BaseOutStructure`.

Reply:
Shouldn't it be `BaseInStructure` for the `*const` cases?

Reply:
That wouldn't be entirely faithful since we can still be assigning to more `p_next` down the line, realistically this should be one long chain of `*mut`?

Reply:
I'm not sure I follow. We only write to `self.p_next` and `next.p_next`, we don't traverse the pointers ourselves.

Reply:
Never mind.

Point is that, if we add a `next = A->B` to an existing chain `C->D->E` (with `C->A->B->D->E` as a result), if `A` were to point to B through `p_next: *const` we would "not be allowed" to assign `D` as next to `B`, since `B` "is const here".

> we don't traverse the pointers ourselves.

In that sense yes, `ptr_chain_iter` definitely does to get hold of `last_next`.

In other words, this whole Vulkan declaration seems a bit flawed and I haven't thus far found (nor really searched for) any reasons for some structs to have this const. That pretty much "forces" you to always build the chain from end to start.

(Before you ask, I'm aware that the above doesn't directly map to writing something _inside_ `self.inner.p_next`, where we do the const cast above. It's trivial to come up with an example that does though, with multiple calls of `push_next`).

Again though, none of this really matters, it's just an oddity and shouldn't hold us back from merging this cleanup. We can look at the `as _` another time if there appears to be some valid reasoning that we should follow through in our API, though I doubt it (most likely it'll only limit the order to prepending).